src/Zumba/Util/JsonSerializer.php

```diff
@@ -43,10 +45,23 @@ class JsonSerializer {
 	protected $preserveZeroFractionSupport;
 
 	/**
+	 * Closure serializer instance
+	 *
+	 * @var SuperClosure\SerializerInterface
+	 */
+	protected $closureSerializer;
+
+	/**
 	 * Constructor.
+	 *
+	 * @param SuperClosure\SerializerInterface $closureSerializer
 	 */
-	public function __construct() {
+	public function __construct(ClosureSerializerInterface $closureSerializer = null) {
 		$this->preserveZeroFractionSupport = defined('JSON_PRESERVE_ZERO_FRACTION');
+
+		if ($closureSerializer !== null) {
```

Comment:
@jrbasso I guess this isn't strictly necessary since you are checking `$this->closureSerializer === null` below.